tests/std/tests/P0401R6_allocate_at_least/test.cpp

```diff
@@ -0,0 +1,93 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <algorithm>
+#include <cassert>
+#include <memory>
+#include <numeric>
+#include <type_traits>
+
+using namespace std;
+
+static_assert(is_standard_layout_v<allocation_result<int*>>);
+static_assert(is_trivially_copyable_v<allocation_result<int*>>);
+
+struct generous_allocator {
+    using value_type = int;

```

Comment:
we need also `pointer` to get the right return type

Reply:
`pointer` is not required and defaults to `value_type*`.

Reply:
Better said, we should have a test where pointer if provided and different to ensure that we return the right thing